lib/topological_inventory/providers/common/operations/source.rb

```diff
@@ -149,16 +167,54 @@ def update_endpoint(status, error_message)
             logger.availability_check("Failed to update Endpoint(ID: #{endpoint.id}) - #{e.message}", :error)
           end
 
+          def update_endpoint_by_kafka(status, error_message)
+            if endpoint.nil?
+              logger.availability_check("Failed to update Endpoint for Source id:#{source_id}. Endpoint not found", :error)
+              return
+            end
+
+            availability_status_message(
+              :resource_type => "endpoint",
```

Comment:
although it'll work, I'll rather see `Endpoint` with big "E" to point out this is polymorphic relation